File: scribbleshare-frontend/src/scripts/protocol/server/messages/ServerMessageCanvasDelete.js
import ServerMessage from "../ServerMessage.js";
import ServerMessageType from "../ServerMessageType.js";
import CanvasDelete from "../../../canvas/CanvasDelete.js";

export default class ServerMessageCanvasDelete extends ServerMessage {
    constructor(reader) {
        super(ServerMessageType.CANVAS_DELETE);
        this.canvasDeletes = [];
        let length = reader.readUint8();
        for (let i = 0; i < length; i++) {
            this.canvasDeletes.push(new CanvasDelete(reader));
        }
    }
}

File: scribbleshare-room/src/main/java/net/stzups/scribbleshare/room/server/websocket/protocol/server/messages/ServerMessageCanvasDelete.java
package net.stzups.scribbleshare.room.server.websocket.protocol.server.messages;

import io.netty.buffer.ByteBuf;
import net.stzups.scribbleshare.data.objects.canvas.CanvasDelete;
import net.stzups.scribbleshare.room.server.websocket.protocol.server.ServerMessage;
import net.stzups.scribbleshare.room.server.websocket.protocol.server.ServerMessageType;

public class ServerMessageCanvasDelete extends ServerMessage {
    private final CanvasDelete[] canvasDeletes;

    public ServerMessageCanvasDelete(CanvasDelete[] canvasDeletes) {
        super(ServerMessageType.CANVAS_DELETE);
        this.canvasDeletes = canvasDeletes;
    }

    public void serialize(ByteBuf byteBuf) {
        byteBuf.writeByte((byte) canvasDeletes.length);
        for (CanvasDelete canvasDelete : canvasDeletes) {
            canvasDelete.serialize(byteBuf);
        }
    }
}
